core/cap-0038.md

```diff
@@ -979,23 +979,40 @@ As noted, for each step in the path the exchange will be routed to the order
 book or liquidity pool that yields the best price for the entire exchange. The
 behavior is changed such that:
 
-1. The price of the exchange is computed for the liquidity pool, recording the
-best price but discarding any changes to the ledger. If the exchange is not
-possible due to insufficient liquidity or the exchange would cause an INT64_MAX
-overflow of either liquidity pool reserve, then this possibility is not
-considered.
-2. The price of the exchange is computed for the order book. If the exchange is
-not possible due to insufficient liquidity, then this possibility is not
-considered.
-3. The result from the order book is returned if the exchange is possible on the
-order book and either
-    a. the exchange is possible on the liquidity pool and the order book yields
-a better price, or
-    b. the exchange is not possible on the liquidity pool.
-4. Otherwise, any changes to the ledger are discarded. If the exchange on the
-liquidity pool is possible then that result is returned.
-5. Otherwise, the operation fails with whatever result would have been returned
-in the absence of the liquidity pool
+1. The price of the exchange is computed for the liquidity pool, or it is
+recorded that the exchange is not possible. There are multiple reasons that the
+exchange might not be possible, including insufficient liquidity or INT64_MAX
+overflow of either pool reserve. Note that exceeding limits set by the operation
+does not qualify as "not possible" in this context.
+2. The price of the exchange is computed for the order book, or it is recorded
+that the exchange is not possible. There are multiple reasons that the exchange
+might not be possible, including insufficient liquidity or a self trade. Note
+that exceeding limits set by the operation does not qualify as "not possible" in
+this context.
+3. If both exchanges are possible, then choose the one which produces the best
+price. In the event that both prices are equal, choose the liquidity pool.
+4. If the exchange is not possible on the liquidity pool, then return whatever
+result was produced when exchanging with the order book.
+
+As an example of the above, consider a path payment strict send with path
+`A -> B -> C:`
+
+    - The `A -> B` conversion is performed successfully
+    - There is insufficient liquidity in the liquidity pool for B and C to
+      perform the `B -> C` exchange
+    - The `B -> C` exchange in the order book results in a self trade
+    - The operation fails with `PATH_PAYMENT_STRICT_SEND_CROSS_SELF`
+
+It is important to recognize that this happens on each step in the path, so
```

Comment:
A similar comment under https://github.com/stellar/stellar-protocol/blob/master/core/cap-0038.md#no-interleaved-execution would be helpful (specifically that the restriction is on a single hop in the path and not the entire path). I think it's possible to mistake no interleaved execution to mean "only use the order book or the liquidity pool for every hop in the path".

Reply:
Done.